chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -529,15 +529,8 @@ func (n *NeutrinoNotifier) historicalConfDetails(confRequest chainntnfs.ConfRequ
 			*blockHash, wire.GCSFilterRegular,
 		)
 		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve regular filter for "+
-				"height=%v: %v", scanHeight, err)
-		}
-
-		// If the block has no transactions other than the Coinbase
-		// transaction, then the filter may be nil, so we'll continue
-		// forward int that case.
-		if regFilter == nil {
-			continue
+			return nil, fmt.Errorf("unable to retrieve regular "+
```

Comment:
Do we also want to include some simple polling logic here until the planned fix is in? Otherwise the historical rescan will fail and won't be retried, requiring a restart of lnd for it to happen again.

Reply:
Yeah, for now wanted to do the minimal change possible to avoid the disaster scenario. Do you think we should add that mitigation here? (the real fix as you say, is a little more advanced)

Reply:
I think so, since this is critical for channel confirmation. Just copying the `pollCFilter` method that's used in `btcwallet` should help a bit, I'd increase the poll interval to something like a second though. We can then remove it once the proper fix is in.

Reply:
Can't we just pass in the required optional arguments for it to _continually_ poll itself inside neutrino? This was the default in the past, but the re-org issues were discovered causing us to remove retries at the query level, so higher level logic could handle awareness of non-existent blocks, etc.

Reply:
Alternatively, can't we just _re use_ the rescan logic here? This is basically a bounded rescan from a known point in the past to now. This would allow us to leverage all the retry and re-org awareness already in that core routine.

Reply:
This is likely a larger change though, as we'll need to construct all its interface dependancies here in the router.

Reply:
Yeah, these are bigger changes. A simple poll loop would be ok to add this late in the game I would say.

Reply:
Retry loop: https://github.com/lightningnetwork/lnd/pull/3593/commits/4c1d50e6782f0cda68e7f0dafa0e7ef78054cf0a

Reply:
> Can't we just pass in the required optional arguments for it to continually poll itself inside neutrino?

We can and it seems to me like the cleaner approach as well.

Reply:
Would also add a TODO to remove the retrying once the proper fix lands.

Reply:
Done.